app/components/upload-app/status/index.ts

```diff
@@ -0,0 +1,177 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { computed } from '@ember/object';
+import ENUMS from 'irene/enums';
+import SubmissionModel from 'irene/models/submission';
+import Store from '@ember-data/store';
+import { inject as service } from '@ember/service';
+import DS from 'ember-data';
+import { addObserver, removeObserver } from '@ember/object/observers';
+import RealtimeService from 'irene/services/realtime';
+import { task } from 'ember-concurrency';
+import IntlService from 'ember-intl/services/intl';
+
+export default class UploadAppStatusComponent extends Component {
+  @tracked showDropdown = false;
+  @tracked anchorRef?: HTMLElement | null;
+  
+
+  @tracked submissions: DS.AdapterPopulatedRecordArray<SubmissionModel> | null =
+    null;
+
+  @service declare store: Store;
+  @service declare realtime: RealtimeService;
+  @service declare uploadApp: any;
+  @service declare intl: IntlService;
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+    this.initialize();
+  }
+
+  @computed('submissions.@each.status')
+  get submissionList() {
+    return this.submissions ? this.submissions : [];
+  }
+
+  @computed('submissions.@each.status', 'uploadedApps.files.length')
+  get submissionNumbers() {
+    let obj = {
+      completedSubmissions: 0,
+      failedSubmissions: 0,
+      runningSubmissions: 0,
+      variant: 'indeterminate',
+      loaderColor: 'green-loader',
+    };
+    if (this.submissions) {
+      this.submissions.map((submission) => {
```

Comment:
use `forEach` and can remove `if` condition by changing to `(this.submissions || [])`